esp_delta_ota/include/esp_delta_ota.h

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache 2.0 License
+ *
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ */
+
+#pragma once
+
+#include "esp_err.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef void *esp_delta_ota_handle_t;
+
+// Callback for reading the source data
+typedef esp_err_t (*src_read_cb_t)(uint8_t *buf_p, size_t size, int src_offset);
+
+// Callback for working on the data generated by applying patch on the source data
+typedef esp_err_t (*merged_stream_write_cb_t)(const uint8_t *buf_p, size_t size);
+
+typedef struct esp_delta_ota_cfg {
+    src_read_cb_t read_cb;                  /*!< Read Callback */
+    merged_stream_write_cb_t write_cb;      /*!< Write Callback */
+} esp_delta_ota_cfg_t;
+
+/**
+ * @brief Initializes the delta OTA process
+ *
+ * @param[in] cfg pointer to esp_delta_ota_cfg_t structure.
+ * @return - NULL   On failure
+ *         - esp_delta_ota_handle_t handle
+ */
+esp_delta_ota_handle_t esp_delta_ota_init(esp_delta_ota_cfg_t *cfg);
+
+/**
+ * @brief This function performs the patch applying operation on the source data.
+ *
+ * @param[in] handle    esp_delta_ota_handle_t handle
+ * @param[in] buf       pointer to patch buffer
+ * @param[in] size      size of patch buffer. This should be atleast 4 bytes during the first call of this API
+ *                      as detools need to process its header during the first call of this API.
+ * @return - ESP_OK
```

Comment:
I implemented support for inputting any amount of bytes in detools, so this restriction should not be needed anymore. It's on master and not yet officially released, but should work just fine. Feel free to give it a try.

Reply:
Hello @eerimoq 
Thanks for adding the support for supplying any number of bytes in `detools`. It works fine and I have updated the PR for the same.